### PR TITLE
Streamline the dev process

### PR DESCRIPTION
**Pull request by rvighne**
This add several small changes to improve the testing and development process. The most important change is that the compiler is now run during `test` to do type-checking (this was already enforced in CI, but didn't run locally unless you asked for it).

We may add more improvements following the meeting on Thursday.

### CODE REVIEW

**Review by yashlala — APPROVED** [review]
We'll have to disable some unicorn rules on a case-by-case basis later (eg. `no-null`), but this is a good starting point.

**Comment by rvighne** [comment]
> We'll have to disable some unicorn rules on a case-by-case basis later (eg. `no-null`), but this is a good starting point.

The only rule we're using from unicorn is `no-array-for-each` (note we're not extending from unicorn, just adding one under the `rules` section).

File: .eslintrc.json [hunk 1]
```diff
@@ -13,28 +13,26 @@
 		"plugin:promise/recommended",
 		"plugin:jest/recommended",
 		"plugin:jest/style",
-		"prettier",
-		"prettier/@typescript-eslint"
```
Comment (Rohanator9000) [inline .eslintrc.json:17]: Why did you remove this?

File: .eslintrc.json [hunk 2]
```diff
@@ -13,28 +13,26 @@
...
-		"tsdoc/syntax": "error",
 		"no-restricted-syntax": [
 			"error",
 			"ForInStatement",
 			"LabeledStatement",
 			"WithStatement"
 		],
 		"no-return-assign": ["error", "except-parens"],
-		"github/array-foreach": ["error"],
+		"unicorn/no-array-for-each": "error",
```
Comment (Rohanator9000) [inline .eslintrc.json:28]: Why did you switch from GitHub to unicorn?

File: package.json [hunk 3]
```diff
@@ -1,29 +1,32 @@
 {
   "name": "junknet",
   "version": "0.1.0",
-  "description": "Zero-config distributed parallel processing",
+  "description": "Zero-config distributed parallel processing.",
   "license": "Apache-2.0",
   "files": [
     "dist",
-    "docs"
+    "!**/*.tsbuildinfo"
```
Comment (Rohanator9000) [inline package.json:8]: We're not using the `--incremental` flag, right? What else would produce `.tsbuildinfo`?

File: .gitignore [hunk 4]
```diff
@@ -1,9 +1,18 @@
 node_modules
+*.tgz
```
[the remark this answers is not in the document]
Comment (rvighne) [reply .gitignore:2]: npm pack

File: .eslintrc.json (hunk 1 above)
Comment (rvighne) [reply .eslintrc.json:17]: Upgraded `esling-config-prettier`, turns out they have removed the second config because it's built in to the first.

File: .eslintrc.json (hunk 2 above)
Comment (rvighne) [reply .eslintrc.json:28]: See the commit log. This is the same type of rule as `github/array-foreach`, but it's auto-fixable.

Also, unicorn has a lot of other interesting rules we may want to use in the future (not enabled now cause I don't want to break existing stuff).

File: package.json (hunk 3 above)
Comment (rvighne) [reply package.json:8]: `composite: true` implies `incremental: true`

**Review by Rohanator9000 — APPROVED** [review]
